hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/RocksDatabase.java

```diff
@@ -523,4 +591,64 @@ public String toString() {
     return name;
   }
 
+  @VisibleForTesting
+  public List<LiveFileMetaData> getSstFileList() {
+    return db.get().getLiveFilesMetaData();
+  }
+
+  /**
+   * return the max compaction level of sst files in the db.
+   * @return level
+   */
+  private int getLastLevel() {
+    return getSstFileList().stream()
+        .max(Comparator.comparing(LiveFileMetaData::level)).get().level();
+  }
+
+  /**
+   * Deletes sst files which do not correspond to prefix
+   * for given table.
+   * @param prefixPairs , a list of pairs (TableName,prefixUsed).
+   * @throws RocksDBException
+   */
+  public void deleteFilesNotMatchingPrefix(
+      List<Pair<String, String>> prefixPairs,
+      BooleanTriFunction<String, String, String, Boolean> filterFunction)
+      throws RocksDBException {
+    for (LiveFileMetaData liveFileMetaData : getSstFileList()) {
+      String sstFileColumnFamily =
+          new String(liveFileMetaData.columnFamilyName(),
+              StandardCharsets.UTF_8);
+      int lastLevel = getLastLevel();
+      for (Pair<String, String> prefixPair : prefixPairs) {
+        String columnFamily = prefixPair.getKey();
+        String prefixForColumnFamily = prefixPair.getValue();
+        if (sstFileColumnFamily.equals(columnFamily)) {
+          // RocksDB #deleteFile API allows only to delete the last level of
+          // SST Files. Any level < last level won't get deleted and
+          // only last file of level 0 can be deleted
+          // and will throw warning in the rocksdb manifest.
+          // Instead perform the level check here
+          // itself to avoid failed delete attempts for lower level files.
+          if (liveFileMetaData.level() == lastLevel && lastLevel != 0) {
+            String firstDbKey = new String(liveFileMetaData.smallestKey(),
+                StandardCharsets.UTF_8);
+            String lastDbKey = new String(liveFileMetaData.largestKey(),
+                StandardCharsets.UTF_8);
+            boolean isKeyWithPrefixPresent = filterFunction
```

Comment:
I did it this way because RocksDatabase is a generic class and we cant add OM Keytable related logic i.e to extract bucketKey from the DBKey etc in this class . Instead the method deletKeysMatchingPrefix takes an argument as a functional interface which any component OM,SCM,DN can leverage by just defining the filter rule by writing own BooleanTriFunction. Not every table would have "/" as delim so adding it in RocksDatabase would make it specific usage. Please let me know your thoughts if this sounds good or if any other approach.